cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -624,17 +624,18 @@ func ValidateIgnorePreflightErrors(ignorePreflightErrorsFromCLI, ignorePreflight
 func ValidateSocketPath(socket string, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
+	if len(socket) == 0 { // static and dynamic defaulting should have added a value to the field already
+		return append(allErrs, field.Invalid(fldPath, socket, "empty CRI socket"))
+	}
+
 	u, err := url.Parse(socket)
 	if err != nil {
 		return append(allErrs, field.Invalid(fldPath, socket, fmt.Sprintf("URL parsing error: %v", err)))
 	}
 
-	if u.Scheme == "" {
-		if !filepath.IsAbs(u.Path) {
-			return append(allErrs, field.Invalid(fldPath, socket, fmt.Sprintf("path is not absolute: %s", socket)))
-		}
```

Comment:
this has been a long running bug, because a path such as `/some/socket` is not absolute when running on Windows. `IsAbs` requires Windows paths to have drive letters.
at the same time the Windows CRI endpoints do not have drive letters because they are named pipes...

if a path is not absolute we just pass it to the kubelet and let it throw errors instead of kubeadm.